File: r/2015/day02a/day02a.r
#!/usr/bin/env Rscript

usage <- function() {
    args <- commandArgs()
    progname <- sub("^--file=", "", args[grep("^--file=", args)])
    cat("usage:", progname, "<input file>\n")
    quit(status = 1)
}

process <- function(filename) {
    totalArea <- 0
    lines <- suppressWarnings(readLines(filename))
    for ( line in lines ) {
        dims <- as.integer(strsplit(line, "x")[[1]])
        l <- dims[1]
        w <- dims[2]
        h <- dims[3]
        area1 <- l * w
        area2 <- l * h
        area3 <- w * h
        surfaceArea <- 2 * area1 + 2 * area2 + 2 * area3
        minArea <- min(area1, area2, area3)
        totalArea <- totalArea + surfaceArea + minArea
    }
    return(totalArea)
}

main <- function() {
    args <- commandArgs(trailingOnly = TRUE)
    if ( length(args) < 1 ) {
        usage()
    }
    filename <- args[1]
    result <- process(filename)
    cat("result =", result, "\n")
}

main()

File: nim/2015/day02a/day02a.nim
import os, sequtils, strutils

proc processFile(filename: string): int =
  for line in lines(filename):
    let dims = (line.split('x').map(parseInt))
    let (l, w, h) = (dims[0], dims[1], dims[2])
    let (area1, area2, area3) = (l * w, l * h, w * h)
    let surfaceArea = 2 * area1 + 2 * area2 + 2 * area3
    let minArea = min(area1, min(area2, area3))
    result += surfaceArea + minArea

when isMainModule:
  if paramCount() < 1:
    quit "usage: " & getAppFilename() & " <input file>", 1

  let filename = paramStr(1)
  try:
    echo "result = ", processFile(filename)
  except OSError as e:
    quit "error while processing file `" & filename & "': " & e.msg, 1
